test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -728,6 +785,96 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
         }
 
         [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_IStringLocalizer()
+        {
+            // Arrange
+            var provider = CreateIStringLocalizerProvider();
+
+            var key = ModelMetadataIdentity.ForType(typeof(EnumWithLocalizedDisplayNames));
+            var attributes = new object[0];
+
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+            provider.CreateDisplayMetadata(context);
+
+            // Act & Assert
+            using (new CultureReplacer("fr-FR", "fr-FR"))
+            {
+                Assert.All(
+                    context.DisplayMetadata.EnumGroupedDisplayNamesAndValues
+                        .Where(kvp => kvp.Value == "2"),
+                    kvp => Assert.Equal("Loc_Two_Name fr-FR", kvp.Key.Name));
```

Comment:
This isn't what I suggested. Why are you checking only a single element of the `EnumGroupedDisplayNamesAndValues` collection?

If you must check just one, do what is done elsewhere e.g.
``` c#
var kvp = Assert.Single(context.DisplayMetadata.EnumGroupedDisplayNamesAndValues, k => k.Value == "2");
Assert.Equal("Loc_Two_Name fr-FR", kvp.Key.Name);
```
Even there, note `==` should almost never be used for `string` comparisons and `Assert.Equal()` helpfully includes an overload that accepts `StringComparer.Ordinal`. Be explicit IOW.